app/models/foreman_deployments/deployment.rb

```diff
@@ -3,10 +3,50 @@ class Deployment < ActiveRecord::Base
     belongs_to :configuration, :class_name => 'ForemanDeployments::Configuration', :autosave => true
 
     validates :name, :presence => true
-    validates :configuration, :presence => true
+    validates :configuration,   :presence => true
+    validates :organization_id, :presence => true, :if => lambda { SETTINGS[:organizations_enabled] }
```

Comment:
Let's move the taxonomy handling into its own concern - it will leave the model cleaner.